encode.go

```diff
@@ -92,34 +110,33 @@ func Encode(dst, src []byte) []byte {
 		if len(src) != 0 {
 			d += emitLiteral(dst[d:], src)
 		}
+		e.cur += 4
 		return dst[:d]
 	}
 
-	// Initialize the hash table. Its size ranges from 1<<8 to 1<<14 inclusive.
-	const maxTableSize = 1 << 14
-	shift, tableSize := uint(32-8), 1<<8
-	for tableSize < maxTableSize && tableSize < len(src) {
-		shift--
-		tableSize *= 2
+	// Ensure that e.cur doesn't wrap.
+	if e.cur > 1<<30 {
```

Comment:
It's not obvious to me what the constraints on e.cur are. Why would it get up to 1<<30 when the tableSize is 1<<14? Why do you then say e.cur = 0 instead of e.cur &= mask?